go-controller/pkg/allocator/udn/ips.go

```diff
@@ -0,0 +1,53 @@
+package udn
+
+import (
+	"fmt"
+	"net"
+
+	idallocator "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/id"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	utilnet "k8s.io/utils/net"
+)
+
+const (
+	masqueradeIPv4IDName = "v4-masquerade-ips"
+	masqueradeIPv6IDName = "v6-masquerade-ips"
+)
+
+// AllocateV4MasqueradeIPs will return a pair of masquerade IPv4 addresses calculated from
+// the networkID argument
+func AllocateV4MasqueradeIPs(networkID int) ([]net.IP, error) {
```

Comment:
There's another `allocateV4MasqueradeIPs()` function in pkg/config/utils.go already.  Should we consider a different name for the new one?  Same for v6.

Reply:
No need to rename since we will qualify the name with udn since the package is `pkg/allocator/udn` 

we will use it like the following
```golang
import(
   udnallocator "github.com/ovn-org/ovn-kubernetes/pkg/allocator/udn"
)

udnallocator.AllocateV4MasqueradeIP

```

Reply:
I know but for old fashioned people like me that still use `git grep` it was a bit confusing. :)

Reply:
an alternative is to put back the Allocator struct, but in thise case free functions are enough I think.

Reply:
OK, sounds good to me.